yaml-tests/src/test/resources/join-tests.yamsql

```diff
@@ -67,16 +67,45 @@ test_block:
     -
       # Get names of people working on a project
       - query: select fname, lname from emp where exists (select * from project where emp_id = emp.id);
+      - supported_version: !current_version
       - unorderedResult: [{"Emily", "Martinez"},
                           {"Daniel", "Miller"},
                           {"Megan", "Miller"}]
+    -
+      # Version of above query to simulate force_continuations with older versions.
+      # Can remove once we do not care about cross-version compatibility before !current_version
+      # See: https://github.com/FoundationDB/fdb-record-layer/issues/3219
+      - query: select fname, lname from emp where exists (select * from project where emp_id = emp.id);
+      - maxRows: 1
+      - initialVersionLessThan: !current_version
+      - result: [{"Emily", "Martinez"}]
+      - result: [{"Daniel", "Miller"}]
+      - result: [{"Daniel", "Miller"}]
+      - result: [{"Megan", "Miller"}]
+      - result: [{"Megan", "Miller"}]
+      - result: []
+      - initialVersionAtLeast: !current_version
     -
       # Get names of people working on a project in Sales department
       - query: select fname, lname from
                   (select fname, lname, dept_id from emp where exists (select * from project where emp_id = emp.id)) as sq,
                   dept
                where sq.dept_id = dept.id and dept.name = 'Sales';
+      - supported_version: !current_version
       - unorderedResult: [{"Daniel", "Miller"}]
+    -
+      # Version of above query to simulate force_continuations with older versions.
+      # Can remove once we do not care about cross-version compatibility before !current_version
+      # See: https://github.com/FoundationDB/fdb-record-layer/issues/3219
+      - query: select fname, lname from
+                  (select fname, lname, dept_id from emp where exists (select * from project where emp_id = emp.id)) as sq,
+                  dept
+               where sq.dept_id = dept.id and dept.name = 'Sales';
+      - maxRows: 1
+      - initialVersionLessThan: !current_version
+      - result: [{"Daniel", "Miller"}]
+      - result: []
```

Comment:
Isn't this the same as the above? It returns 1 row with `{"Daniel", "Miller"}`?

Reply:
Oh, so the key is that if you had, e.g. `4.1.8.0 then !currentVersion` it would return something like:
```
- result: [{"Daniel", "Miller"}]
- result: [{"Daniel", "Miller"}]
- result: []
```

Perhaps a comment in the other branch then?

Reply:
That's what would happen if the initial version were `!current_version` and the second version were 4.1.8.0. I've added some comments about that in those branches. This is, however, sort of a general pattern, which maybe will become clearer when we try to merge my `enforce-force-continuations` branch in, and I'm not sure what the right balance here is between documenting everything and making everything a little unwieldy given that the same sorts of changes are applied multiple places.